logstash-core/lib/logstash/api/modules/logging.rb

```diff
@@ -0,0 +1,51 @@
+# encoding: utf-8
+#
+java_import org.apache.logging.log4j.core.LoggerContext
+
+module LogStash
+  module Api
+    module Modules
+      class Logging < ::LogStash::Api::Modules::Base
+        # retrieve logging specific parameters from the provided settings
+        #
+        # return any unused configurations
+        def handle_logging(settings)
+          Hash[settings.map do |key, level|
+            if key.start_with?("logger.")
+              _, path = key.split("logger.")
+              LogStash::Logging::Logger::configure_logging(level, path)
+              nil
+            else
+              [key, level]
+            end
+          end]
+        end
+
+        put "/" do
+          begin
+            req_body = LogStash::Json.load(request.body.read)
```

Comment:
IIRC, since request.body.read it's an IO object and might have been read already by a before hook or rack handler it is recommended to rewind it before consuming it.

see http://www.sinatrarb.com/intro.html
```ruby
post "/api" do
  request.body.rewind  # in case someone already read it
  data = JSON.parse request.body.read
  "Hello #{data['name']}!"
end
```